src/silo/preprocessing/preprocessor.cpp

```diff
@@ -0,0 +1,515 @@
+            database_config.schema.partition_by.value()
+         );
+      } else {
+         partition_by_select = "0 as partition_id";
+         partition_by_where = "";
+      }
+
+      for (const auto& [seq_name, _] : reference_genomes.raw_nucleotide_sequences) {
+         auto return_code = preprocessing_db.query(fmt::format(
+            "create or replace view nuc_{0} as\n"
+            "select {1} as key, nuc_{0} as sequence,"
+            "{3}"
+            "{2} \n"
+            "from preprocessing_table, partition_key_to_partition "
+            "{4};",
+            seq_name,
+            database_config.schema.primary_key,
+            order_by_select,
+            partition_by_select,
+            partition_by_where
+         ));
+
+         if (return_code->HasError()) {
+            SPDLOG_ERROR(return_code->GetError());
+            throw PreprocessingException(return_code->GetError());
+         }
+      }
+
+      for (const auto& [seq_name, _] : reference_genomes.raw_aa_sequences) {
+         auto return_code = preprocessing_db.query(fmt::format(
+            "create or replace view gene_{0} as\n"
+            "select {1} as key, gene_{0} as sequence, "
+            "{3}\n"
+            "{2} \n"
+            "from preprocessing_table, partition_key_to_partition "
+            "{4};",
+            seq_name,
+            database_config.schema.primary_key,
+            order_by_select,
+            partition_by_select,
+            partition_by_where
+         ));
+
+         if (return_code->HasError()) {
+            SPDLOG_ERROR(return_code->GetError());
+            throw PreprocessingException(return_code->GetError());
+         }
+      }
+   } else {
+      std::string order_by_select = ", partitioned_metadata." + database_config.schema.primary_key +
+                                    " as " + database_config.schema.primary_key;
+      if (database_config.schema.date_to_sort_by.has_value()) {
+         order_by_select += ", partitioned_metadata." +
+                            database_config.schema.date_to_sort_by.value() + " as " +
+                            database_config.schema.date_to_sort_by.value();
+      }
+
+      for (const auto& [seq_name, reference_sequence] :
+           reference_genomes.raw_nucleotide_sequences) {
+         preprocessing_db.generateNucSequenceTable(
+            "raw_nuc_" + seq_name,
+            reference_sequence,
+            preprocessing_config.getNucFilenameNoExtension(seq_name).replace_extension(
+               silo::preprocessing::FASTA_EXTENSION
+            )
+         );
+
+         auto return_code = preprocessing_db.query(fmt::format(
+            R"-(
+create or replace view nuc_{0} as
+select key, sequence,
+partitioned_metadata.partition_id as partition_id
+{2}
+from raw_nuc_{0} raw, partitioned_metadata
+where raw.key = partitioned_metadata.{1};)-",
+            seq_name,
+            database_config.schema.primary_key,
+            order_by_select
+         ));
```

Comment:
```suggestion
        string raw_table_name = "raw_nuc_" + seq_name;
         preprocessing_db.generateNucSequenceTable(
            raw_table_name,
            reference_sequence,
            preprocessing_config.getNucFilenameNoExtension(seq_name).replace_extension(
               silo::preprocessing::FASTA_EXTENSION
            )
         );

         auto return_code = preprocessing_db.query(fmt::format(
            R"-(
create or replace view nuc_{0} as
select key, sequence,
partitioned_metadata.partition_id as partition_id
{2}
from {raw_table_name} raw, partitioned_metadata
where raw.key = partitioned_metadata.{1};)-",
            seq_name,
            database_config.schema.primary_key,
            order_by_select
         ));
```
We could extract `"raw_nuc_" + seq_name` to make the connection more obvious